Created the method Run() and the majority of the functions that need …

paulaOliverosHollis opened the pull request:
…to be called in this method.

The intro and menu are succesfully printing to the screen and the IsUserChoiceValid is succesflly checking for errors.

Commit:
Created the method Run() and the majority of the functions that need to be called in this method.

The intro and menu are succesfully printing to the screen and the IsUserChoiceValid is succesflly checking for errors.

zhollis21 commented on Class1.cs:
```diff
@@ -0,0 +1,59 @@
+﻿using System;
```
This file needs to be deleted... It looks like a duplicate of DeveloperGameClass

You deleted it from your solution, but that doesn't remove it from the folder in your file system.

Try to carefully look over each file you are going to check in before checking it in.

paulaOliverosHollis replied:
OK, I see. I'm working on it right now!

paulaOliverosHollis replied:
I fixed this issue.

zhollis21 replied:
@paulaOliverosHollis There is only 1 commit on this pull request, so I'm not sure if you forgot to push or if you checked it into the wrong place. Please update this on your "CreatingDeveloperGameClass"

zhollis21 replied:
![image](https://user-images.githubusercontent.com/43797367/62722908-70081680-b9dd-11e9-853a-a058a3bdc2dc.png)

paulaOliverosHollis replied:
Please check now.

zhollis21 replied:
Yep, I see it now.

zhollis21 replied:
Please check the pull request to make sure it is updated before saying it's fixed. :)

zhollis21 commented on DeveloperGame/DeveloperGame/DeveloperGameClass.cs:
```diff
@@ -0,0 +1,118 @@
+﻿using System;
+
+public class DeveloperGameClass
```
Don't put "Class" in the name of a class. This is true when creating almost any type

paulaOliverosHollis replied:
I fixed this issue.

zhollis21 commented on DeveloperGame/DeveloperGame/DeveloperGameClass.cs:
```diff
@@ -0,0 +1,118 @@
+    {
+        PrintIntro();
+
+        while (true)
```
This is good, but I think the best solution would be:

```
char userMenuChoice;
do
{
    userMenuChoice = GetUserMenuChoice();

    ... Do the rest of your if's except for the quit because it is in the while loop

} while (!UserChoseToQuit);
```

I think it makes it very clear about when we stop our while loop. 
Before the while(true) could be deceiving

paulaOliverosHollis replied:
I fixed this issue.

zhollis21 commented on DeveloperGame/DeveloperGame/DeveloperGameClass.cs:
```diff
@@ -0,0 +1,118 @@
+            {
+                Play();
+            }
+            if (UserChoseToSeeLeaderBoard(userMenuChoice))
```
While it is true that UserChoseToPlay() and UserChoseToSeeLeaderBoard() should never be true on the same character, as another programmer I had to go and look at these functions to see that they were mutually exclusive.

To make it easier to read exclusive choices, please use if, else if, and else so we know only one can happen just by looking at this code, without having to step in.

paulaOliverosHollis replied:
I fixed this issue.

zhollis21 commented on DeveloperGame/DeveloperGame/DeveloperGameClass.cs:
```diff
@@ -0,0 +1,118 @@
+
+        while (!IsUserMenuChoiceValid(userMenuChoice))
+        {
+            Console.WriteLine("Sorry, the option you entered is not valid.Please try again!");
```
You need a space before your "Please" it is right up against the period

paulaOliverosHollis replied:
I fixed this issue.

zhollis21 commented on DeveloperGame/DeveloperGame/DeveloperGameClass.cs:
```diff
@@ -0,0 +1,118 @@
+        {
+            return false;
+        }
+        else if (UserChoseToPlay(userMenuChoice[0]) || UserChoseToSeeLeaderBoard(userMenuChoice[0])|| UserChoseToQuit(userMenuChoice[0]))
```
Instead of doing this _else if_ and _else_, just return what you have in your else if. What you have right now is basically:
_"If **this** is true return true, else return false"_

But it would be simpler just to say "return **this**"

paulaOliverosHollis replied:
I fixed this issue.

zhollis21 commented on DeveloperGame/DeveloperGame/DeveloperGameClass.cs:
```diff
@@ -0,0 +1,118 @@
+
+    private bool UserChoseToPlay(char userMenuChoice)
+    {
+        if (userMenuChoice == 'P' || userMenuChoice == 'p')
```
Same issue as above, you are saying _"if **this** is true, return true, else return false"_, just _"return **this**"_

paulaOliverosHollis replied:
I fixed this issue.

zhollis21 commented on DeveloperGame/DeveloperGame/DeveloperGameClass.cs:
```diff
@@ -0,0 +1,118 @@
+
+    private bool UserChoseToSeeLeaderBoard(char userMenuChoice)
+    {
+        if (userMenuChoice == 'L' || userMenuChoice == 'l')
```
Same issue as above, you are saying "if this is true, return true, else return false", just "return this"

paulaOliverosHollis replied:
I fixed this issue.

zhollis21 commented on DeveloperGame/DeveloperGame/DeveloperGameClass.cs:
```diff
@@ -0,0 +1,118 @@
+
+    private bool UserChoseToQuit(char userMenuChoice)
+    {
+        if (userMenuChoice == 'Q' || userMenuChoice == 'q')
```
Same issue as above, you are saying "if this is true, return true, else return false", just "return this"

paulaOliverosHollis replied:
I fixed this issue.

zhollis21 commented on DeveloperGame/DeveloperGame/DeveloperGameProgram.cs:
```diff
@@ -0,0 +1,14 @@
+    {
+        static void Main()
+        {
+            DeveloperGameClass gameOne = new DeveloperGameClass();
```
I don't think "gameOne" is a very good variable name

paulaOliverosHollis replied:
What other names do you suggest?

zhollis21 replied:
"game" or "techGame" or "quizGame" or anything else would probably be fine, but "gameOne" doesn't make much sense as it sounds like this variable will only let the user play once or something like that.

Commit:
Since I changed the game concept, I have also changed the name of the project and class.

I fixed all of the issues in my pull request review as well as the suggestions mentioned.

zhollis21 reviewed (approved):
Looks good! Nice job, very clean code!
